velox/functions/sparksql/SplitFunctions.cpp

```diff
@@ -134,10 +135,127 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
               .build()};
 }
 
+class StringToMap final : public exec::VectorFunction {
+ public:
+  StringToMap(char entryDelim, char keyValueDelim)
+      : entryDelim_(entryDelim), keyValueDelim_(keyValueDelim) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    DecodedVector* inputString = decodedArgs.at(0);
+
+    BaseVector::ensureWritable(
+        rows, MAP(VARCHAR(), VARCHAR()), context.pool(), result);
+    exec::VectorWriter<Map<Varchar, Varchar>> resultWriter;
+    resultWriter.init(*result->as<MapVector>());
+
+    context.applyToSelectedNoThrow(rows, [&](vector_size_t row) {
+      folly::F14FastSet<StringView> keys;
```

Comment:
You probably don't want to create new map for each row. Consider moving outside of the loop and clearing for each row.